### DISCUSSION TITLE
Climate Hackathon

### DISCUSSION THREAD

**Discussion by petermr** [petermr/petermr#37] [Announcements]
# Climate Hackathon ("Learning by Doing")

_This is for visitors to the event and our project. Editors should use links where possible._

## project links:
* [Landing page (Sparkboard)](https://hackathon-piaget.sparkboard.com/public)
* [The program](https://hackathon-piaget.sparkboard.com/project/62a765bd952014003d9ff042)
* [Semantic Climate page: "Learning by Doing" ]( https://hackathon-piaget.sparkboard.com/project/62a327036c6ee3003da01352 )

## suggested activities:
Based on a sample chapter (Chapter04)
### read Executive Summary
* read An Executive Summary or FAQs for a chapter in the IPCC Climate report. (example: "Mitigation, near to mid-term ..." 
 - [full PDF]( https://github.com/petermr/semanticClimate/blob/main/ipcc/ar6/wg3/Chapter04/fulltext.pdf )
 - [full HTML download]( https://github.com/petermr/semanticClimate/blob/main/ipcc/ar6/wg3/Chapter04/fulltext.flow.html) (need to download) 
 - [full HTML view](http://htmlpreview.github.io/?https://github.com/petermr/semanticClimate/blob/main/ipcc/ar6/wg3/Chapter04/fulltext.flow.html)
 - [Executive Summary](https://github.com/petermr/semanticClimate/blob/main/ipcc/ar6/wg3/Chapter04/executive.html) 
 - [View](http://htmlpreview.github.io/?https://github.com/petermr/semanticClimate/blob/main/ipcc/ar6/wg3/Chapter04/executive.html)
### read FAQs. (Chapter04)
 - [Download](https://github.com/petermr/semanticClimate/blob/main/ipcc/ar6/wg3/Chapter04/faq.html)
 - [View]( http://htmlpreview.github.io/?https://github.com/petermr/semanticClimate/blob/main/ipcc/ar6/wg3/Chapter04/faq.html)

## search public science
* choose a climate term (e.g. NDC)
* search literature using `pygetpapers` [install instructions](https://pygetpapers.readthedocs.io/en/latest/)
* annotate/analyse your search results with [`docanalysis`](https://github.com/petermr/docanalysis/blob/main/README.md)

## section and annotate Chapters (very exploratory) - for adventurous hackers
We are developing tools continually and we may have versions that can be run

## creating dictionaries - for adventurous hackers
We have several exploratory tools.

**Comment by ShweataNHegde** [comment]
Pitch presentation [[PDF](https://github.com/petermr/semanticClimate/blob/main/outreach/piaget_hackathon/piaget_hackathon_v4.pdf)] [[PPT](https://github.com/petermr/semanticClimate/blob/main/outreach/piaget_hackathon/piaget_hackathon_v3.pptx)]
## Specific sub-projects 
_We choose 1-2 of these for the hackathon today_
1. Develop a working prototype for a chapter
    -  build dictionary 
    - search and annotate HTML 
2. Test the existing tools. `pygetpapers` and possibly, `docanalysis`.
3. Brainstorm whether a project like ours -- which focuses on solving real problems e.g., climate change -- can be introduced in schools/universities as a learning module. If so, how?
## Expected outcomes
By the end, we hope to
1.  Streamline pipeline
    - learn about the open packages that exist in Python
    - get experts (both climate and tech) feedback on the tools and project
2.  Gain feedback -- improve our documentation and help
3.  Gain insight on
    - how we can take our tech to classrooms
    - how solving real problems can help students learn skills (practising educator's opinions would be valuable)
    - whether if exposing students to global challenges this way is useful
## Challenges
1. Code development always takes more time than you think!
2. Insufficient documentation; OS-specific problems
3. None I can think of! 

Please feel free to add your thoughts in the Reply section and I will add to this Comment.

**Comment by ShweataNHegde** [comment]
2022-06-17 -- Hackathon Meeting 
## Participants
- Peter
- Ximena
- Jiawei
- Nicola
- Rajashri
- Piyush

## Sub-projects
1. Prepare a Google Colab Learning Module
    - Turn Google Colab into a learning module 
    -  Integrate our tech into the curriculum - teachers learn with children
    -  Translate the module to different languages -- Spanish, Italian, etc.
    - Make interactive videos to demonstrate our tools within the module
2. Test and run our software (tech people)

## Questions
1. Are Learning Modules a useful resource to have?
2. How do we get our the initial barrier that people might face when they are first used to something like Google Colab?

**Comment by ShweataNHegde** [comment]
## Updates
- I've created the learning module in English: https://colab.research.google.com/drive/1QtPcwmFWkeE28adnbatBTJ85glzL4kZ-?usp=sharing. Feedback welcome! I've also added a small code snippet that lets us look up Wikidata. 
- Jiawei, Ximena, and Nicola are translating the learning module into Mandarin, Spanish and Italian respectively. Nicola might possibly do French. Rajashri has translated to Bengali! Yay!
- Peter, Rajashri, and Piyush are testing whether the code actually runs and does what they are supposed to do. They have also been able to install `pygetpapers` locally. Great!
- Ximena is particularly interested in reaching out to policymakers! In her words, "_I am a member of the "Network of Educational Policy Specialists in Latin America"(https://red.iiep.unesco.org/es/) I have been helping there - and I am in contact with all the government members of the Educational Ministries - we could create a plan to spread this Hackton we are working in a massive way_"
- Piyush, on the other hand, is keen on coming up with a GUI. In his words, _" we can create a front-end of this for non-technical people as our final end-user program for education purposes_"
- Helene summed up how most of us felt _"...feeling useful is definitely one of the best feelings in the world"_

**Comment by ShweataNHegde** [comment]
UPDATE: 

We have translations in French, Spanish, Italian, Mandarin, Bengali and Hindi! We can now reach out to so many teachers with our tools! Thank you so much for all the volunteers!

**Comment by ShweataNHegde** [comment]
English: https://colab.research.google.com/drive/1QtPcwmFWkeE28adnbatBTJ85glzL4kZ-?usp=sharing - Shweata

Hindi: https://colab.research.google.com/drive/1vp_EBMn8b72UueiZvdSruguYO7fFmXD7?usp=sharing - Piyush

Bengali: https://colab.research.google.com/drive/1AqdHHt90dfABfHL7IEBB6Nc_Ez595QkB - Rajarshi Mondal 

Spanish: https://colab.research.google.com/drive/15SHegKN5H5lpprWLH7jwZ26ol9_TloTo#scrollTo=XOHxBYbU8sWr - Xemina

Italian: https://colab.research.google.com/drive/1rBbaZuIBSz0ESVWVfSb1eRNrDVb9dn1U?usp=sharing - Nicola

Mandarin: https://colab.research.google.com/drive/1HO5-zZcx9w2zkvpwsACxcDIqfM-bcuPV?usp=sharing - Jiawei

French: https://colab.research.google.com/drive/1NFFnIdcoA32bWxVTz5oPbavp6I0W3jr1?usp=sharing - Helene

English Demo: https://youtu.be/B3-CuKmgN5A - Piyush

Summary video (Xemina): https://drive.google.com/drive/folders/18qnd0zVEuVRNvOClcJh17PLjpwwUp4q4?usp=sharing

**Comment by ShweataNHegde** [comment]
So, WE WON! Thanks to all the participants and team for contributing to the Challenge! Also, thanks to the organizers for running such a unique hackathon.

We have been asked to come up with yet another pitch that details the next steps. Let's use this thread to develop ideas for the next steps. 

As Peter pointed out in our daily meeting today (2022-06-22), our goals are to: 
- develop open tech to liberate knowledge from IPCC Reports
- reach out to the world to make sure people
    - discover our tech
    - try it
    - find value 
    - incorporate it in their work practice -- teaching, researching, etc.
    - spread the word

We will continue to develop tech. Some other ideas that align with the goals:
- Run workshops to introduce our tech (targeted towards young people without a formal background in tech)
- Work with climate experts to make our tools more valuable to them
- Conduct surveys to analyse how much young people are aware of IPCC (or climate change) and tech (Python, open databases, etc). Decide on the outreach strategy based on the survey


2 SDGs that our project focuses on are: 
- Climate Action
- Quality Education

**Reply by ShweataNHegde** [reply]
We might also want to collaborate with organizations to reach more young people. 

### [UNESCO MGIEP](https://mgiep.unesco.org/)
- They have a course on climate change: https://mgiep.unesco.org/course-climate-change
- They recently hosted a Twitter Space on Mental Health and Climate Action: https://twitter.com/UNESCO_MGIEP/status/1538776013177356288
### [ Science Gallery Bengaluru](https://bengaluru.sciencegallery.com/)
- https://carbon.scigalleryblr.org/

We should curate all that we can offer in one place, before approaching anyone.

**Reply by ShweataNHegde** [reply]
Some more ideas!

### 1. Develop Climate Change Game
Last year, we had planned Aroma the Game (https://github.com/petermr/CEVOpen/wiki/Aroma-the-Game). We can develop a similar game for young people with climate as a focus. We turn running our Text Data Mining Tools into a Game. People liberate knowledge from climate-related scientific papers (or IPCC Reports) using our tools and dictionaries. This will prompt us to develop tools that people can reuse. The Game can be run either virtually or in person.

### 2. Crowdsource Dictionary Creation and Annotation Verification
Our tools do a great job of annotating, but it isn't perfect. That's where humans step in. We build or make use of existing crowdsourcing platforms (Wikidata?). 

Similar initiative: https://www.nature.com/articles/nindia.2019.61 (I remember attending a workshop of theirs. It was on how to read scientific literature. )
### 3. Create resources
- Video tutorials for our tools
- Podcasts featuring experts (climate, education, tech) and team members

These resources help people discover our tech and try them. 

## Update from 2022-06-27's meeting
### 4. Build a website for our project (Kaartik)
Although all our work is open, there is not one place people to learn about our work. 

### 5. Build a UI for the tools (Manny)

**Reply by ShweataNHegde** [reply]
#### Clipping the next steps Peter outlined in an email
* use it to raise more funds. I don't think this is the top option as it requires a large commitment to run something but it's how some of the non-profit orgs have grown.
* create a seminal object (book, video, podcasts,etc.) using professional input/tools where appropriate
* outreach. Run events.
* work with organization/s . This may involve travel. 
* develop our current tools/resources for reuse by others. This would have to involve specific objectives.

**Reply by ShweataNHegde** [reply]
Think about 
- Who the team might be
- Which organizations can involve

**Comment by petermr** [comment]
YES!!

The first step should be to brainstorm what we'd like to happen, and then
decide which of the ideas is tractable. We can be creative.

On Wed, Jun 22, 2022 at 2:52 PM Shweata N. Hegde ***@***.***>
wrote:

> So, WE WON! Thanks to all the participants and team for contributing to
> the Challenge!
>
> We have been asked to come up with yet another pitch that details the next
> steps. Let's use this thread to develop ideas for the next steps.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/petermr/petermr/discussions/37#discussioncomment-3002479>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFTCS2QXVSTMTHGB33TTELVQMLCFANCNFSM5Y6CDUCA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Peter Murray-Rust
Founder ContentMine.org
and
Reader Emeritus in Molecular Informatics
Dept. Of Chemistry, University of Cambridge, CB2 1EW, UK

**Comment by ShweataNHegde** [comment]
We are meeting Jose, possibly on Tuesday. 
Some questions: 
1. Any strict timeline?
2. Can you describe some projects that you support at SDG Accelerator? 
3. To what extent should the next steps be aligned with the Hackathon Challenge?
4. What details must go into the post-Hackathon pitch?

**Comment by ShweataNHegde** [comment]
https://gt-initiative.org/education-programs/accelerator/
## The Project
1. What is the name of the project?
2. Who are the people already involved in the project? (name, institution, email, role in the project)
## The Problem
1. What is the problem you are trying to solve? Use metrics where possible: for example approximately how many people are affected, and how the problem impacts their lives. _Make climate knowledge accessible to citizens_
3. What SDGs is this project tackling? What targets and indicators are relevant? _Climate Action_ 
## The Solution
### Tell us in a few lines how your solution would address the problem.

### What other approaches have been tried before? What makes your solution better?
### In an ideal case, what is the expected impact of your project within 5 years?
### What stage is your project at? What are the major achievements to date?
### What geographical areas will benefit from your solution?
## SDG Data
### What types of data are you consuming or producing in this project?
### How are these data related to SDG targets or indicators?
### How will the project share the data it generates?
## Impact and Risk
### How will you measure and evaluate the impact of the project?
### Who are the key partners to maximize the impact of your solution?
### How do you plan to sustain and increase the impact of your project?
### What are the major risks in this project and how would you mitigate them?
## Needs and budgets
### What are the main needs preventing your project from moving forward?
### What types of resources would you need to cover those needs?

**Comment by ShweataNHegde** [comment]
We are coming up with the project plan, here: https://docs.google.com/presentation/d/1VeKAJzu07jr0dM2AUd67EUvFYtEi13cd/edit?usp=sharing&ouid=102162640834446538146&rtpof=true&sd=true. Feedback welcome!

**Comment by petermr** [comment]
This is developing nicely.
I talked with my Shuttleworth mentor Karien yesterday; we agreed that
travel and events are very good candidates for spending. They are
unavoidable. (Other unavoidables are web-hosting, domain-name). Do you need
cloud hosting for web services and/or data?

Emphasize the volunteer, chapter-based approach. I like this. I like
"Chapter Champions".

"Locked in PDFs are information that saves the world" I misread this;
suggest rephrasing it.

P.

On Wed, Jul 6, 2022 at 5:48 AM Shweata N. Hegde ***@***.***>
wrote:

> We are coming up with the project plan, here:
> https://docs.google.com/presentation/d/1VeKAJzu07jr0dM2AUd67EUvFYtEi13cd/edit?usp=sharing&ouid=102162640834446538146&rtpof=true&sd=true
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/petermr/petermr/discussions/37#discussioncomment-3089387>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFTCS6D2H7FJLLQJX2P7TTVSUFZLANCNFSM5Y6CDUCA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Peter Murray-Rust
Founder ContentMine.org
and
Reader Emeritus in Molecular Informatics
Dept. Of Chemistry, University of Cambridge, CB2 1EW, UK

**Comment by ShweataNHegde** [comment]
Final presentation pitch to Geneva: https://youtu.be/0MGXic-uEQY. You'll also find the link to slides and Demo in the video description. Feedback welcome.

**Reply by ShweataNHegde** [reply]
You can download and view the presentation from this video, here: https://github.com/petermr/semanticClimate/blob/main/outreach/piaget_hackathon/post_hackathon_pitch_v3.pptx

You can also try our tools and generate the Wordcloud that I showed in the presentation on Google Colab: https://colab.research.google.com/drive/1sT2Die3pV3dLcyHgwZBg3IxS2FJ_8W0-?usp=sharing

**Comment by petermr** [comment]
# Ideas for future (next 12 months)

Shweata  is now developing the SDG-Accelerator (Geneva) IPCC project as a set of workshops , virtual and hybrid.  The approximate timescale (changes weekly!) is something like:

* 2022 Q3 (now) . proof of concepts and prototype intern video. Target. A 1-day online workshop package (Carpentry-like, see below) where participants (singly or in teams) take a single IPCC chapter and make it semantic. Currently we are working on WG3 but assume it translates to WG1 and WG2 as well.
* 2022 Q4 2 or more virtual hackathons to stress test the material. Certainly one with Delhi. This would be great for TiB to help stress-test
* 2023-Q1 March . Real-life (and virtual) event in NIPGR New Delhi. This would be part of a larger bioinformatics event celebrating India's 75th year of Independence. Current plan to have Delhi-based students to help on the ground (e.g. installing, testing, managing conference admin, etc.) 
* 2023-Q2 Europe. (This will depend heavily on her studies). Definitely Cambridge, Possibly Geneva (haven't discussed this but makes sense) and maybe TiB.

Format of the workshop.
I see this as very similar to the Capentries (https://carpentries.org/) - too early to be part of this actual program. I went to Greg Wilson's session in 2013 and it changed my way of thinking and then my life. The method was then:
* You go to a Software Carpentry workshop
* You offer to help with a future repeat event and learn to be an instructor
* You qualify as instructor and run events.

The model a decade ago was that organizations (e.g. libraries or other training) sponsored the event (e.g. paying fares for presenter/s, managing the venue). I can see a similar model here where organizations such as libraries sponsored 1-day events.

Our main limitation at present is we aren't formally knowledgeable on climate, so we'd need to find experts to join in.

The model itself is now much more reliable because we can use cloud computing (Shweata pioneered this in Google Colabs). That means a single set of tested software. The actual goal is less clear.
* we take the IPCC reports as a given corpus, rather like religious texts or a canon of Shakespeare/Goethe, etc.
* we make them semantic automatically and make them readable with hyperlinks, Wikidata, etc.
* we add analytical tools such as statistics, search engines, data display and analysis

We can then offer this to the world and try to market it.

When we have something impressive I will try to find colleagues in Cambridge who will find it useful. It's a solution looking for problems to solve. That's always hard.

**Comment by petermr** [comment]
# **September 2022**

@GY proposes a hackday "in Delhi"  maybe with possible sponsors.  She will provide colleagues to build on our existing  progress. This will be exploratory but should be well prepared with good example material. Suggestions:
* each chapter we should automate
- creation of HTML
- extraction of main sections for each
- extraction of all images
- extraction of image captions (harder)
- extraction of table captions.
- pyamiimage to get word lists
- abbreviations

* @petermr will build an iterator over all chapters
* communal dictionary

videos - on the various stages

The main purpose is to show the breadth of tools for one chapter and scale it automatically. 

**Please can everyone think about what they did for their chapters and try to reduce each traffic-light to a single command.**

**Comment by petermr** [comment]
# form of hackathon
**(NOT final, being edited, please contribute)**

We don't know the capabilities of the delegates (it may be invite-only which will be a big advantage). but we're assuming there will several who have little experience of working online and they will need guides.  We are hoping to have at least 6 helpers.

This is a hackathon,  NOT a competition! There's no set goal or outcome. You can suggest other activities or goals. Different people will find different things of value to contribute and take away. One thing that is always valuable is documentation, so recording what you have done and your impressions,  is really valuable. You'll often work in a mini-team, maybe in a Breakout room. Your social skills will be valuable in making these a success. 

## programme
The programme will be roughly split into two parts: **core** and **options** . Currently we suggest:
### overview of programme (BRIEF)
* why are we doing this?
  - the climate crisis (very brief)
  - IPCC reports (what they are, what they look like)
 
 
* what we are going to do
  - the programme
  - breakout strategy
* the technology
  - Zoom 
  - breakouts and discussions
  - Etherpad for communal ideas and recording
  - Jupyter Notebook / Google Colabs
  - commandline
  - Python
  - running programs

### short introductions?? 
Really important, especially since delegates will bring valuable ideas and resources. Even 15 seconds per person takes 10 minutes for the group and the switch overs between people can also be considerable. Even 20 minutes here is a lot) This needs very careful orchestration. Maybe get everyone to wave initially. (There can also be microphone delays). [see  pre-preparation #delegates] 
* 

# delegates

## registration details
Delegates should pre-prepare a **brief** 1/4 page about themselves. Maybe Etherpad? (NOT part of programme, but we may highlight some issues)
  - who they are
  - where they are from
  - relevant background (academic, industry NGO, govt, high school student, etc.)
  - relevant skills (scripting, editing, HTML, Wikimedia, Python, etc.) 
  - what they want to get from the workshop


## core
We suggest several components: **core** and 3-4 **options**. **Everyone will do the core**. This will take variable amounts of time depending on previous experience, and is not a race! If you find many new challenges, then understanding them, and recording them for others is hugely valuable. 

* **core** will be guided demos and discussions; Much of **core** will be pre-prepared interactive demos, with active helpers.
* we expect that everyone will finish the core, if necessary through helpers

## options
Then several options, some more guided than others
* **exploration**.  modifying existing scripts (e.g. changing chapter or dictionary)
*  **advanced**. is free-er exploration of our tools, e.g. creating a new dictionary or wordcloud. Maybe modifying a Jupyter Notebook. Interfacing with Wikimedia tools. 
* **outreach**. Delegates discuss how they might take these tools to a wider community. One output could be "white-papers"; or modified
*  **subjects** - e.g. agriculture   

# Hackathon technology
## 
Zoom

Anywhere from 4 to 8 Zoom rooms (5-10 people in each)

# Helpers
* on the ground or online in NIPGR
  - 4 interns (Abhi, Abhishek, Kavita, Priti) 
  - GY
* remote
  - Shweata
  - PMR
  -  **Alumni**. We should get as many as possible to attend: Ananya, Ambreen?

# Goals

## to make the IPCC reports accessible to everyone 

  - human annotation, especially in a variety of languages
  - machine annotation (abbreviations,  dictionaries)
  - sectioning (chapters, subchapters , FAQs...)
  - indexing
  - hyperlinking
  - languages and synonyms

 ## to feed our work into other initiatives
   - Wikipedia and Wikidata
   - delegates initiatives
   - 
 ## extract data automatically and build new data

   - tables 
  - figures 
  - entities from fulltext
  - 
## meet other people!


# Outcomes
Success comes in many forms. 


## delegate excitement

 We cannot predict the detail.
* "This looks really useful, can we have a follow-up for my colleagues"
* "This has great relevance to our course on SDGs for India"
* "Could we use the tools to analyze CGIAR reports?"
* "Is this suitable for Master's projects?"
* can we join the next group of interns? what do we need?"
*

## refining the workshop and approach

* what worked and what didn't
* capture action points for future workshops

### material. 
Is it understandable? relevant? 

### procedures. 
* Did the timetable work for delegates?
* Was the balance right?
  - prepared presentations
  - breakout discussions
  - self-paced exercises
  - mini-teams
  - 

### tech
Can it be installed? does it work?

### additions?




# core 

### assumptions (please edit)
* delegates want to see our community, its practices  and values, and the potential for the approach in (a) climate (b) other Indian contexts. 
* delegates can use Zoom at some level (I assume this is likely to be the commonest). (Other tech such as Discord are more complex and frightening). 
* Asynchronous comms is through Zoom (not Slack, not email, not social media). (Zoom is not permanent so 
* Delegates will be interested in our community technology. We don't and we can give them demos:
 - Slack
 - 

I think a large part of the meeting could be "guided"

**Comment by kavita9597** [comment]
12/08/2022

Daily meeting topics covered
1) Understanding zoom breakout rooms
2) Solving Priti's problem about running collab code chunk via Jupyter Notebook

**Reply by petermr** [reply]
Thanks!
